sigh/lib/assets/resign.sh

```diff
@@ -499,43 +515,13 @@ function resign {
     # Replace embedded provisioning profile with new file
     cp -f "$NEW_PROVISION" "$APP_PATH/embedded.mobileprovision"
 
-    #if the current bundle identifier is different from the new one in the provisioning profile, then change it.
-    if [ "$CURRENT_BUNDLE_IDENTIFIER" != "$BUNDLE_IDENTIFIER" ];
-    then
-        log "Updating the bundle identifier from '$CURRENT_BUNDLE_IDENTIFIER' to '$BUNDLE_IDENTIFIER'"
-        PlistBuddy -c "Set :CFBundleIdentifier $BUNDLE_IDENTIFIER" "$APP_PATH/Info.plist"
-        checkStatus
-    fi
-
-    # Update the version number properties in the Info.plist if a version number has been provided
-    if [ "$VERSION_NUMBER" != "" ];
-    then
-        CURRENT_VERSION_NUMBER=$(PlistBuddy -c "Print :CFBundleVersion" "$APP_PATH/Info.plist")
-        if [ "$VERSION_NUMBER" != "$CURRENT_VERSION_NUMBER" ];
-        then
-            log "Updating the version from '$CURRENT_VERSION_NUMBER' to '$VERSION_NUMBER'"
-            PlistBuddy -c "Set :CFBundleVersion $VERSION_NUMBER" "$APP_PATH/Info.plist"
-            PlistBuddy -c "Set :CFBundleShortVersionString $VERSION_NUMBER" "$APP_PATH/Info.plist"
-        fi
-    fi
+    change_plist_value "$INFO_PLIST" "CFBundleIdentifier" "$BUNDLE_IDENTIFIER"
 
-    # Update short version string in the Info.plist if provided
-    if [[ -n "$SHORT_VERSION" ]];
-    then
-        CURRENT_VALUE="$(PlistBuddy -c "Print :CFBundleShortVersionString" "$APP_PATH/Info.plist")"
-        # Even if the old value is same - just update, less code, less debugging
-        log "Updating the short version string (CFBundleShortVersionString) from '$CURRENT_VALUE' to '$SHORT_VERSION'"
-        PlistBuddy -c "Set :CFBundleShortVersionString $SHORT_VERSION" "$APP_PATH/Info.plist"
-    fi
+    change_plist_value "$INFO_PLIST" "CFBundleVersion" "$VERSION_NUMBER"
+    change_plist_value "$INFO_PLIST" "CFBundleShortVersionString" "$VERSION_NUMBER"
 
-    # Update bundle version in the Info.plist if provided
-    if [[ -n "$BUNDLE_VERSION" ]];
-    then
-        CURRENT_VALUE="$(PlistBuddy -c "Print :CFBundleVersion" "$APP_PATH/Info.plist")"
-        # Even if the old value is same - just update, less code, less debugging
-        log "Updating the bundle version (CFBundleVersion) from '$CURRENT_VALUE' to '$BUNDLE_VERSION'"
-        PlistBuddy -c "Set :CFBundleVersion $BUNDLE_VERSION" "$APP_PATH/Info.plist"
-    fi
+    change_plist_value "$INFO_PLIST" "CFBundleVersion" "$SHORT_VERSION"
```

Comment:
This may look a little weird, to set after setting before, but the validation steps in the beginning of the script will restrict that VERSION_NUMBER will not be set if SHORT_VERSION and BUNDLE_VERSION are set.